spring-beans/src/main/java/org/springframework/beans/factory/BeanFactoryUtils.java

```diff
@@ -73,7 +80,7 @@ public abstract class BeanFactoryUtils {
 	 * @see BeanFactory#FACTORY_BEAN_PREFIX
 	 */
 	public static boolean isFactoryDereference(@Nullable String name) {
-		return (name != null && name.startsWith(BeanFactory.FACTORY_BEAN_PREFIX));
+		return (name != null && !name.isEmpty()	&& name.charAt(0) == FACTORY_BEAN_PREFIX);
```

Comment:
```suggestion
		return (name != null && !name.isEmpty() && name.charAt(0) == FACTORY_BEAN_PREFIX);
```

That was a tab instead of a space, which breaks the build.

In the future, please run a full local build before submitting a PR.

Thanks!

No need to update the PR: I'm fixing that locally while merging.